src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParser.cs

```diff
@@ -40,7 +40,9 @@ static TdsParser()
             // For a default installation of SqlServer the encoding exchanged during Login is 1252. This encoding is not loaded by default
             // See Remarks at https://msdn.microsoft.com/en-us/library/system.text.encodingprovider(v=vs.110).aspx 
             // SqlClient needs to register the encoding providers to make sure that even basic scenarios work with Sql Server.
+#if !MONO
             Encoding.RegisterProvider(CodePagesEncodingProvider.Instance);
+#endif
```

Comment:
Rather than using a MONO ifdef, can we instead move the whole static cctor to its own partial file (e.g. TdsParser.RegisterEncoding.cs), which would be included in the .NET Core build but not in Mono's build?  That has the added benefit for Mono of not having a static cctor so the type can be beforefieldinit, and IMO has the added benefit for Core of keeping this code (which is only necessary because of some distribution choices elsewhere in the platform) out of the main TdsParser implementation.